pkg/universal-cnf/vppagent/backend_test.go

```diff
@@ -0,0 +1,188 @@
+package vppagent
+
+import (
+	"os"
+	"testing"
+	"github.com/stretchr/testify/assert"
+
+	"go.ligato.io/vpp-agent/v3/proto/ligato/vpp"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	connectioncontext "github.com/networkservicemesh/networkservicemesh/controlplane/api/connectioncontext"
+	"github.com/networkservicemesh/networkservicemesh/sdk/common"
+	interfaces "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/interfaces"
+	vpp_l3 "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/l3"
+)
+
+func TestBuildVppIfNameNSCPeer(t *testing.T) {
+
+	b := UniversalCNFVPPAgentBackend{}
+	podName := "helloworld-ucnf-78956df47b-r5fj4"
+	conn := &connection.Connection{
+		Labels: map[string]string {
+			"podName": podName,
+		},
+	}
+
+	expected := podName
+	got := b.buildVppIfName("", "", conn)
+
+	assert.Equal(t, expected, got)
+}
+
+func TestBuildVppIfNameVL3NSEPeer(t *testing.T) {
+
+	b := UniversalCNFVPPAgentBackend{}
+	podName := "helloworld-ucnf-78956df47b-r5fj4"
+	conn := &connection.Connection{
+		Labels: map[string]string {
+			"ucnf/peerName": podName,
+		},
+	}
+
+	expected := podName
+	got := b.buildVppIfName("", "", conn)
+
+	assert.Equal(t, expected, got)
+}
+
+func TestBuildVppIfNameDefault(t *testing.T) {
+
+	serviceName := "ucnf"
+	defaultIfName := "endpoint0"
+
+	b := UniversalCNFVPPAgentBackend{
+		EndpointIfID: map[string]int {},
+	}
+
+	conn := &connection.Connection{}
+
+	expected := defaultIfName + "/0"
+	got := b.buildVppIfName(defaultIfName, serviceName, conn)
+
+	assert.Equal(t, expected, got)
+}
+
+func TestProcessEndpoint(t *testing.T) {
+
+	const (
+		srcIpAddr = "192.168.22.1"
+		srcIpRoute = "192.168.0.1/16"
+		workspaceEnv = "workspace"
+	)
+
+	podName := "helloworld-ucnf-78956df47b-r5fj4"
+	b := UniversalCNFVPPAgentBackend{}
+	vppconfig := &vpp.ConfigData{}
+	serviceName := "ucnf"
+	ifName := "endpoint0"
```

Comment:
I would suggest that podName, workspaceEnv, serviceName, ifName/defaultIfName, and other variables to be set as a const () in this file, after imports. I see they are used in multiple places. And srcIpAddr and srcIpRoute into scrIpAddrEndpoint/Client maybe

Reply:
Sure, will do! Thanks for the review.